axelrod/strategies/axelrod_second.py

```diff
@@ -1454,3 +1454,55 @@ def strategy(self, opponent: Player) -> Action:
         self.flack *= 0.5  # Defections have half-life of one round
 
         return random_choice(1.0 - self.flack)
+
+
+class Leyvraz(Player):
+    """
+    Strategy submitted to Axelrod's second tournament by Fransois Leyvraz
+    (K68R) and came in eleventh in that tournament.
+
+    The strategy uses the opponent's last three moves to decide on an action
+    based on the following ordered rules.
+
+    1. If opponent Defected last two turns, then Defect with prob 75%.
+    2. If opponent Defected three turns ago, then Cooperate.
+    3. If opponent Defected two turns ago, then Defect.
+    4. If opponent Defected last turn, then Defect with prob 50%.
```

Comment:
I think the docstrings don't match the description:

```
(D, D, D): 0.5
```

That doesn't correspond to ` If opponent Defected last turn, then Defect with prob 50%.` as indicated for example by `(D, C, D): 1.0`?

Reply:
Apologies: I see you said "ordered" rules.

Reply:
You're right.  I was hoping the tests would catch any issues.  I'll go through this more carefully and re-run the fingerprints.

Reply:
No sorry apologies again, that's still not exact then, according to the ordered description: `(D, D, D)` should defect with prob 75%.

Now that we have this `dict`, we could almost just describe the strategy by saying:

```
Depending on the 3 most recent actions by the opponent the strategy cooperates with probability:

- If (C, C, C): probability 1;
- If (C, C, D): probability 0.5;
...
```

and essentially write out the 8 cases that correspond to the `dict` (which indeed really helps with readability here :+1:).

Reply:
I think (D, D, D) should be 0.25, by the first rule.  (These rules get applied in order.)  These others seem okay; (D, C, D) is 1.0 by rule 2.

Reply:
I think you are correct:

```python
>>> import axelrod as axl
>>> import axelrod_fortran as axlf
>>> players = (axl.Defector(), axlf.Player("k68r"))
>>> match = axl.Match(players, turns=1000)
>>> results = match.play()
>>> match.cooperation()
(0,261)
```

so `k68r` is cooperating 25% of the time against `D, D, D`.

Reply:
Whereas your current branch:

```
>>> players = (axl.Defector(), axl.Leyvraz())
>>> match = axl.Match(players, turns=1000)
>>> axl.seed(0)
>>> results = match.play()
>>> match.cooperation()
(0, 508)
```

Reply:
A direct map of the previous implementation:

```
-        if recent_history[-1] == D and recent_history[-2] == D:
 -            return random_choice(0.25)
 -        if recent_history[-3] == D:
 -            return C
 -        if recent_history[-2] == D:
 -            return D
 -        if recent_history[-1] == D:
 -            return random_choice(0.5)
 -        return C  # recent_history == [C, C, C]
```

gives:

```
{
   (C, D, D): 0.25,
   (D, D, D): 0.25,
   (D, C, C): 1,
   (D, C, D): 1,
   (D, D, C): 1,
   (C, D, C): 0,
   (C, C, D): 0.5,
   (C, C, C): 1
}
```

which I believe gives the same as you have, apart from, as you say `(D, D, D)`.

Would be good to confirm the fps just in case :+1: